src/main/java/athena/timetable/Timetable.java

```diff
@@ -81,18 +90,216 @@ private void populateTimetable() {
+            row += String.format(" %s|", shortenOrPadString(taskInfoWriter.apply(task), boxWidth));
+        }
+
+        return row;
+    }
+
+    /**
+     * Generates the first row for a day in the timetable, which contains the name of the day and the task names.
+     *
+     * @param day       The TimetableDay object for this day.
+     * @param startHour The starting hour in 24-hour representation.
+     * @param endHour   The ending hour in 24-hour representation.
+     * @return A string containing name of the day and the task names.
+     */
+    private String drawTimetableDayFirstRow(TimetableDay day, int startHour, int endHour) {
+        String dayShortName = day.getDate().getDayOfWeek().toString().substring(0, 3).toUpperCase();
+        String row = "|  " + dayShortName + "  |";
+        row += drawTimetableDayRow(day, startHour, endHour, new Function<Task, String>() {
+            @Override
+            public String apply(Task task) {
+                return task.getName();
+            }
+        });
+        row += "\n";
+        return row;
+    }
+
+    /**
+     * Generates the second row for a day in the timetable, which contains the date of the day and the task numbers.
+     *
+     * @param day       The TimetableDay object for this day.
+     * @param startHour The starting hour in 24-hour representation.
+     * @param endHour   The ending hour in 24-hour representation.
+     * @return A string containing date of the day and the task numbers.
+     */
+    private String drawTimetableDaySecondRow(TimetableDay day, int startHour, int endHour) {
+        LocalDate date = day.getDate();
+        String row = String.format("| %s/%s |", date.getDayOfMonth(), date.getMonthValue());
+
+        row += drawTimetableDayRow(day, startHour, endHour, new Function<Task, String>() {
+            @Override
+            public String apply(Task task) {
+                return String.format("[%d]", task.getNumber());
+            }
+        });
+
+        row += "\n";
+        return row;
+    }
+
+    /**
+     * Generates the row representing a day in the timetable.
+     *
+     * @param day       The TimetableDay object for this day.
+     * @param startHour The starting hour in 24-hour representation.
+     * @param endHour   The ending hour in 24-hour representation.
+     * @return A string containing information about a day in the timetable.
+     */
+    private String drawTimetableDay(TimetableDay day, int startHour, int endHour) {
+        String row = "";
+        row += drawTimetableDayFirstRow(day, startHour, endHour);
+        row += drawTimetableDaySecondRow(day, startHour, endHour);
+        row += drawBottomBorder(startHour, endHour);
+        return row;
+    }
+
+    private LocalDate getFirstDayOfWeek() {
+        LocalDate now = LocalDate.now();
+        TemporalField field = WeekFields.of(Locale.forLanguageTag("en_SG")).dayOfWeek();
+        return now.with(field, 1);
+    }
```

Comment:
Something similar is in 115, could consider making it an attribute or a method call from the Task/Time class

Reply:
It turns out that the one in 115 was looking for the first date that matches the day, e.g. given Wednesday return 21 Oct 2020. So it's different from this. But yea I agree it would be better for this to be somewhere else, so that it can also be commonly used by the date related tests.